source/module_hamilt_lcao/hamilt_lcaodft/operator_lcao/deepks_lcao.cpp

```diff
@@ -347,25 +337,46 @@ void hamilt::DeePKS<hamilt::OperatorLCAO<TK, TR>>::calculate_HR()
         std::vector<int> trace_alpha_row;
         std::vector<int> trace_alpha_col;
         std::vector<double> gedms;
-        int ib=0;
-        for (int L0 = 0; L0 <= orb.Alpha[0].getLmax();++L0)
+        if(!GlobalC::ld.get_if_equiv())
         {
-            for (int N0 = 0;N0 < orb.Alpha[0].getNchi(L0);++N0)
+            int ib=0;
+            for (int L0 = 0; L0 <= orb.Alpha[0].getLmax();++L0)
             {
-                const int inl = GlobalC::ld.get_inl(T0, I0, L0, N0);
-                const double* pgedm = GlobalC::ld.get_gedms(inl);
-                const int nm = 2*L0+1;
-        
-                for (int m1=0; m1<nm; ++m1) // m1 = 1 for s, 3 for p, 5 for d
+                for (int N0 = 0;N0 < orb.Alpha[0].getNchi(L0);++N0)
                 {
-                    for (int m2=0; m2<nm; ++m2) // m1 = 1 for s, 3 for p, 5 for d
+                    const int inl = GlobalC::ld.get_inl(T0, I0, L0, N0);
+                    const double* pgedm = GlobalC::ld.get_gedms(inl);
```

Comment:
I would suggest passing the GlobalC::ld as an input parameter

Reply:
I've already expalined the reason why GlobalC is used here, and as the original file contains tons of GlobalC already, it seems to me that this would be the responsibility of whoever re-constructed this part of the code.